primitives/arithmetic/src/traits.rs

```diff
@@ -302,3 +307,519 @@ pub trait SaturatedConversion {
 	}
 }
 impl<T: Sized> SaturatedConversion for T {}
+
+/// Arithmetic operations with safe error handling.
+///
+/// This module provide a readable way to do safe arithmetics, turning this:
+///
+/// ```
+/// # use sp_arithmetic::{traits::EnsureSub, ArithmeticError};
+/// # fn foo() -> Result<(), ArithmeticError> {
+/// # let mut my_value: i32 = 1;
+/// # let other_value: i32 = 1;
+/// my_value = my_value.checked_sub(other_value).ok_or(ArithmeticError::Overflow)?;
+/// # Ok(())
+/// # }
+/// ```
+///
+/// into this:
+///
+/// ```
+/// # use sp_arithmetic::{traits::EnsureSubAssign, ArithmeticError};
+/// # fn foo() -> Result<(), ArithmeticError> {
+/// # let mut my_value: i32 = 1;
+/// # let other_value: i32 = 1;
+/// my_value.ensure_sub_assign(other_value)?;
+/// # Ok(())
+/// # }
+/// ```
+///
+/// choosing the correct [`ArithmeticError`] it should return in case of fail.
+///
+/// The *EnsureOps* family functions follows the same behavior as *CheckedOps* but
+/// returning an [`ArithmeticError`] instead of `None`.
+///
+/// [`ArithmeticError`]: crate::ArithmeticError
```

Comment:
I get a doc error without this. I think it makes sense to have it because `ArithmeticError` is not used in the `traits` module itself, only in the `ensure` module.

Reproduced with `cargo doc -p sp-arithmetic`

Reply:
Added a new suggestion, so that we don't need this here anymore.